### PR TITLE
Add some Resources to resourcemanager

### PR DESCRIPTION
**Pull request by fengzhou-msft**
Add PolicyAssignment, PolicyDefinition, PolicySetDefinition, DataPolicyManifest, PolicyExemption, ManagementLock, ResourceLink to resourcemanager.

This PR is based on autogenerated code, but some manual work was made:
1. Removed CreatedByType.cs under PsuedoGenerated/Resources/Models folder.
2. Added get containers methods in Subscription, ManagementGroup, Tenant, ArmResource, ArmClient.
3. Updated ResourceLinkContainer GetAll method to take in a scope parameter and remove the overridden ValidateResourceType method so the container can only be created with a tenant. It's moved udner Custom folder.
4. Add `Get(All)BuiltInPolicy(Set)Definitions` methods in Tenant instead of exposing the `GetPolicy(Set)Definitions` methods there since tenant does not have create/delete methods for `Policy(Set)Definition`.

Readme used to generate the code:
https://github.com/fengzhou-msft/autorest.csharp/blob/more_samples/samples/Azure.ResourceManager.CoreResources/readme.md

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs [hunk 1]
```diff
@@ -141,5 +141,41 @@ protected async Task<IEnumerable<Location>> ListAvailableLocationsAsync(Resource
...
+        #endregion
+
+        #region ResourceLink
+        /// <summary> Gets an object representing a ResourceLinkContainer along with the instance operations that can be performed on it. </summary>
+        /// <returns> Returns a <see cref="ResourceLinkContainer" /> object. </returns>
+        public virtual ResourceLinkContainer GetResourceLinks()
+        {
+            return new ResourceLinkContainer(this);
+        }
+        #endregion
```
Comment (fengzhou-msft) [inline sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs:179]: @Yao725 let's remove `GetResourceLinks()` here and only put it in `Tenant.cs`.

File: sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/Tenant.cs [hunk 2]
```diff
@@ -251,5 +258,278 @@ public virtual SubscriptionContainer GetSubscriptions()
...
         }
+
+        /// <summary>
+        /// Gets the data policy manifest container for this tenant.
+        /// </summary>
+        /// <returns> A container of the data policy manifest. </returns>
+        public virtual DataPolicyManifestContainer GetDataPolicyManifests()
+        {
+            return new DataPolicyManifestContainer(this);
+        }
```
Comment (fengzhou-msft) [inline sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/Tenant.cs:269]: @Yao725 you need to delegate this method in ArmClient, like this one:
https://github.com/Azure/azure-sdk-for-net/blob/fef0b61dbce12d7372f4af1bd47ed7be5fe805dd/sdk/resourcemanager/Azure.ResourceManager/src/ArmClient.cs#L341

File: sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+        {
+            _clientDiagnostics = new ClientDiagnostics(ClientOptions);
+            _restClient = new ResourceLinksRestOperations(_clientDiagnostics, Pipeline, BaseUri);
+        }
+
+        /// <summary> Verify that the input resource Id is a valid container for this type. </summary>
+        /// <param name="identifier"> The input resource Id to check. </param>
+        protected override void ValidateResourceType(ResourceIdentifier identifier)
+        {
+        }
```
Comment (fengzhou-msft) [inline sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs:44]: @Yao725 as discussed, let's remove this method.

File: sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary> Gets a list of resource links at and below the specified source scope. </summary>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <returns> A collection of <see cref="ResourceLink" /> that may take multiple service requests to iterate over. </returns>
+        public virtual Pageable<ResourceLink> GetAll(CancellationToken cancellationToken = default)
```
Comment (fengzhou-msft) [inline sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs:266]: @Yao725 as discussed, let's add a `ResourceIdentifier scope` parameter.

Comment (Yao725) [reply sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs]: added

File: sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs (hunk 3 above)
Comment (Yao725) [reply sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/ResourceLinkContainer.cs]: removed

File: sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs (hunk 1 above)
Comment (Yao725) [reply sdk/resourcemanager/Azure.ResourceManager/src/ArmResource.cs]: removed

File: sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/Tenant.cs (hunk 2 above)
Comment (Yao725) [reply sdk/resourcemanager/Azure.ResourceManager/src/PsuedoGenerated/Resources/Tenant.cs:269]: done

**Pull request closed without merging**